crates/amalthea/src/language/shell_handler.rs

```diff
@@ -31,7 +29,7 @@ pub trait ShellHandler: Send {
     async fn handle_info_request(
         &mut self,
         req: &KernelInfoRequest,
-    ) -> Result<KernelInfoReply, Exception>;
+    ) -> crate::Result<KernelInfoReply>;
```

Comment:
Hmmmmmmmmmm

I actually disagree with this strongly enough to request that we talk about this one first.

This change implies that rather than encoding that we return a well-defined `Exception`, we now return an arbitrary `amalthea::Error`. That is not what the jupyter spec declares as the possible return value here.

---

Oh hmm. I see now that what you are doing in `handle_request()` is taking that `amalthea::Error` and mapping it into the appropriate `Exception` and then you send that out.

I thought for sure we instead wanted to return something like this `KernelInfoReply` here:

```rust
pub enum KernelInfoReply {
    Ok(KernelInfoReplyData),
    Error(ReplyException),
}

pub struct KernelInfoReplyData {
    /// The execution status ("ok" or "error")
    pub status: Status,

    /// Version of messaging protocol
    pub protocol_version: String,

    /// Information about the language the kernel supports
    pub language_info: LanguageInfo,

    /// A startup banner
    pub banner: String,

    /// Whether debugging is supported
    pub debugger: bool,

    /// A list of help links
    pub help_links: Vec<HelpLink>,
}

pub struct ReplyException {
    /// The name of the exception
    pub ename: String,

    /// The value/description of the exception
    pub evalue: String,

    /// List of traceback frames, as strings
    pub traceback: Vec<String>,
}
```

Where `ReplyException` is reused across reply types, and `ExecuteReplyException` is a special variant with the execution count.